### PR TITLE
CI: move to GitHub Actions for release

### PR DESCRIPTION
**Pull request by rachelcarmena**
## Problems

* Failures from Travis CI. For instance:
> The job exceeded the maximum time limit for jobs, and has been terminated.
* [It deploys though build fails](https://travis-ci.org/arrow-kt/arrow/builds/605196796?utm_source=github_status&utm_medium=notification):
```
The command "./gradlew clean build" exited with 1.
...
The command "./deploy-scripts/deploy.sh" exited with 0.
```
* The Travis CI badge is showing the status of the last execution from any branch (not just `master`)
Screenshot from 2019-11-06 17-46-58

## Solution

We're using GitHub Actions sucessfully in Arrow Meta, so the proposal is moving to GitHub Actions step by step.

In this pull request, we're removing Travis CI which is being used for deployments right now (to OSS repository and Bintray repository)

## Changes

* Remove `.travis.yml` file.
* New script: `.github/workflows/release.yml`
* New badge for release status. By default, it displays the status of the default branch: `master`.
* No need for build status (it's included in release status).
* Improvement: avoid duplicated executions for `push` and `pull_request` events.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      env:
+        JAVA_OPTS: -Xms512m -Xmx1024m
+      run: |
+        ./gradlew clean build
+    - name: Publish artifacts
+      env:
+        BINTRAY_USER: ${{ secrets.BINTRAY_USER }}
+        BINTRAY_API_KEY: ${{ secrets.BINTRAY_API_KEY }}
+      run: |
+        ./gradlew uploadArchives
```
Comment (nomisRev) [inline .github/workflows/release.yml]: Are we not using `./deploy-scripts/deploy.sh` anymore like we did for Travis?

Comment (rachelcarmena) [reply .github/workflows/release.yml]: Thanks for the review @nomisRev :+1: We simplified that part in Arrow Meta when including what those scripts were doing. In this way, we consider that it's easier to know what's happening in GitHub Actions. Thanks!

Comment (nomisRev) [reply .github/workflows/release.yml]: Typing on phone... Opening & closing... Sorry :D

Awesome! That's what I was hoping for.

Thanks for the update.

Comment (rachelcarmena) [reply .github/workflows/release.yml]: Nothing to apologize, reviews are always useful, mate!! Thanks to you!! :raised_hands:

**Pull request merged**